app/assets/javascripts/discourse/lib/to-markdown.js.es6

```diff
@@ -0,0 +1,211 @@
+import parseHTML from 'discourse/helpers/parse-html';
+
+String.prototype.trimLeft = function() {
+    return this.replace(/^\s+/,"");
+}
+
+const Tag = Ember.Object.extend({
+
+  decorate(text) {
+    if (this.prefix || this.suffix) {
+      return [this.prefix, text, this.suffix].join("");
+    }
+
+    return text;
+  },
+
+  innerMarkdown() {
+    let text = this.element.innerMarkdown();
+
+    if (!["li"].includes(this.element.name)) {
+      text = text.replace(/^ +/g, "");
+    }
+
+    return text;
+  },
+
+  toMarkdown() {
+    const text = this.innerMarkdown();
+
+    if (text && text.trim()) {
+      return this.decorate(text);
+    }
+
+    return text;
+  }
+});
+
+Tag.reopenClass({
+
+  heading(name, prefix) {
+    return Tag.extend({
+      name: name,
+      prefix: `\n\n${prefix} `,
+      suffix: "\n\n",
+    });
+  },
+
+  emphasis(name, decorator) {
+    return Tag.extend({name: name, prefix: decorator, suffix: decorator});
```

Comment:
With ES6, instead of `name: name` you can just do `name`. This should also be done on L54-55 for `name` and `text`.